src/Microsoft.Identity.Abstractions/DownstreamApi/IDownstreamApi.HttpMethods.cs

```diff
@@ -765,5 +769,756 @@ public partial interface IDownstreamApi
+        ///         myItem,
+        ///         options =>
+        ///         {
+        ///           options.RelativePath = $"api/todolist/{myItem.Id}";
+        ///         });
+        /// </code>
+        /// </example>
+        public Task PutForAppAsync<TInput>(
+            string? serviceName,
+            TInput input,
+            JsonTypeInfo<TInput> inputJsonTypeInfo,
+            Action<DownstreamApiOptionsReadOnlyHttpMethod>? downstreamApiOptionsOverride = null,
+            CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Calls, using <see cref="HttpMethod.Put"/>, a downstream API with some input data and returning data.
+        /// By default the input data is serialized in JSON and the returned data is deserialized from JSON but you can provide your own serializer and your own deserializer in the action
+        /// you pass-in through the <paramref name="downstreamApiOptionsOverride"/> parameter.
+        /// </summary>
+        /// <typeparam name="TInput">Generic input type.</typeparam>
+        /// <typeparam name="TOutput">Generic output type.</typeparam>
+        /// <param name="inputJsonTypeInfo">JSON serialization metadata for TInput</param>
+        /// <param name="outputJsonTypeInfo">JSON serialization metadata for TOutput</param>
+        /// <param name="serviceName">Name of the service describing the downstream API. There can
+        /// be several configuration named sections mapped to a <see cref="DownstreamApiOptions"/>,
+        /// each for one downstream API. You can pass-in null, but in that case <paramref name="downstreamApiOptionsOverride"/>
+        /// needs to be set.</param>
+        /// <param name="input">Data sent to the downstream web API, through the body or the HTTP request.</param>
+        /// <param name="downstreamApiOptionsOverride">[Optional] Overrides the options proposed in the configuration described
+        /// by <paramref name="serviceName"/>.</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns> The value returned by the downstream web API.</returns>
+        /// <example>
+        /// <code>
+        ///  var result = await _downstreamApi.PutForAppAsync&lt;MyItem, IEnumerable&lt;MyItem&gt;&gt;(
+        ///         "MyService",
+        ///         myItem,
+        ///         options =>
+        ///         {
+        ///           options.RelativePath = $"api/todolist/{myItem.Id}";
+        ///         });
+        /// </code>
+        /// </example>
+        public Task<TOutput?> PutForAppAsync<TInput, TOutput>(
+            string? serviceName,
+            TInput input,
+            JsonTypeInfo<TInput> inputJsonTypeInfo,
+            JsonTypeInfo<TOutput> outputJsonTypeInfo,
+            Action<DownstreamApiOptionsReadOnlyHttpMethod>? downstreamApiOptionsOverride = null,
+            CancellationToken cancellationToken = default)where TOutput : class;
+
+#if NETSTANDARD2_1_OR_GREATER
```

Comment:
I'm not sure this #if makes sense`. You are already in a "if .NET 8 or greater" condition. This code won't be used in a netstandard2.1 compilation.

Reply:
This is now removed inside the net8 block